query-graphs-js/src/pathTree.ts

```diff
@@ -353,7 +365,9 @@ export class PathTree<TTrigger, RV extends Vertex = Vertex, TNullEdge extends nu
 
     const localSelections = this.mergeLocalSelectionsWith(other);
     const newChilds = this.childs.concat(other.childs);
-    return new PathTree(this.graph, this.vertex, localSelections, this.triggerEquality, newChilds);
+    const contextToSelection = (this.contextToSelection || other.contextToSelection) ? new Map([...(this.contextToSelection || []), ...(other.contextToSelection || [])]) : null;
+    const parameterToContext = (this.parameterToContext || other.parameterToContext) ? new Map([...(this.parameterToContext || []), ...(other.parameterToContext || [])]) : null;
```

Comment:
It looks like the only difference between `mergeMapOrNull()` and this logic, is that when only one map is non-null, this logic copies the map while `mergeMapOrNull()` returns the original. Is the copying necessary/does later logic mutate this? If we don't ever mutate these maps in-place, it looks like we can use `mergeMapOrNull()` here.